libsplinter/src/admin/service/event/store/diesel/mod.rs

```diff
@@ -17,3 +17,35 @@
 
 mod models;
 mod schema;
+
+#[cfg(all(test, feature = "admin-service-event-store-diesel", feature = "sqlite"))]
+pub mod tests {
+    use crate::migrations::run_sqlite_migrations;
+
+    use diesel::{
+        r2d2::{ConnectionManager, Pool},
+        sqlite::SqliteConnection,
+    };
+
+    /// Creates a connection pool for an in-memory SQLite database with only a single connection
+    /// available. Each connection is backed by a different in-memory SQLite database, so limiting
+    /// the pool to a single connection ensures that the same DB is used for all operations.
+    fn create_connection_pool_and_migrate() -> Pool<ConnectionManager<SqliteConnection>> {
+        let connection_manager = ConnectionManager::<SqliteConnection>::new(":memory:");
+        let pool = Pool::builder()
+            .max_size(1)
+            .build(connection_manager)
+            .expect("Failed to build connection pool");
+
+        run_sqlite_migrations(&*pool.get().expect("Failed to get connection for migrations"))
+            .expect("Failed to run migrations");
+
+        pool
+    }
+
+    #[test]
+    /// Test that the sqlite migrations can be run successfully
+    fn test_event_store_migrations() {
```

Comment:
Technically we don't need to add this test for every store because all migrations always get run.